proxy/proxy.go

```diff
@@ -209,6 +213,12 @@ func (p *T) AsyncProduce(topic string, key, message sarama.Encoder) {
 // available for consumption. In that case the user should back off a bit
 // and then repeat the request.
 func (p *T) Consume(group, topic string, ack Ack) (consumer.Message, error) {
+	p.consumerMu.RLock()
```

Comment:
With this change `p.consumerMu` mutex is made to wrap `p.eventsChMapMu`. That may result in higher contention on the former mutex. So rather then grab mutex and check for `p.consumer == nil` I suggest checking for `p.cfg.Consumer.Disabled == true`, since the config is immutable you do not need to use locks at all.

Reply:
Makes sense - fixed.